pico_set_binary_type(binary blocked_ram) causes link failure

I want to use Pico's SRAM in unstriped configuration so that I can manage who accesses which bank manually. However, using `pico_set_binary_type(blink blocked_ram)` causes linking to fail with the message below (here I modified the blink example with the single additional line for demonstration and easy reproduction).

> [  0%] Linking CXX executable blink.elf
ERROR: Memory segment 21000a4c->21000d14 is outside of valid address range for device
make[2]: *** [blink/CMakeFiles/blink.dir/build.make:677: blink/blink.elf] Error 253
make[2]: *** Deleting file 'blink/blink.elf'
make[1]: *** [CMakeFiles/Makefile2:3535: blink/CMakeFiles/blink.dir/all] Error 2
make: *** [Makefile:84: all] Error 2

The address range rejected by the linker seems to be exactly the place code should be, so I suspect the linker is just misconfigured.

To answer my own question... I get this when using the `master` branch of pico-sdk:
```
[100%] Linking CXX executable blink.elf
ERROR: Memory segment 21000a50->21000d1c is outside of valid address range for device
blink/CMakeFiles/blink.dir/build.make:671: recipe for target 'blink/blink.elf' failed
make[3]: *** [blink/blink.elf] Error 253
make[3]: *** Deleting file 'blink/blink.elf'
CMakeFiles/Makefile2:3550: recipe for target 'blink/CMakeFiles/blink.dir/all' failed
make[2]: *** [blink/CMakeFiles/blink.dir/all] Error 2
CMakeFiles/Makefile2:3557: recipe for target 'blink/CMakeFiles/blink.dir/rule' failed
make[1]: *** [blink/CMakeFiles/blink.dir/rule] Error 2
Makefile:157: recipe for target 'blink' failed
make: *** [blink] Error 2
```

and this when using the `develop` branch of pico-sdk:
```
[100%] Linking CXX executable blink.elf
ERROR: Memory segment 21000230->210005f8 is outside of valid address range for device
blink/CMakeFiles/blink.dir/build.make:671: recipe for target 'blink/blink.elf' failed
make[3]: *** [blink/blink.elf] Error 253
make[3]: *** Deleting file 'blink/blink.elf'
CMakeFiles/Makefile2:3571: recipe for target 'blink/CMakeFiles/blink.dir/all' failed
make[2]: *** [blink/CMakeFiles/blink.dir/all] Error 2
CMakeFiles/Makefile2:3578: recipe for target 'blink/CMakeFiles/blink.dir/rule' failed
make[1]: *** [blink/CMakeFiles/blink.dir/rule] Error 2
Makefile:157: recipe for target 'blink' failed
make: *** [blink] Error 2
```

--- a/src/host/CMakeLists.txt
+++ b/src/host/CMakeLists.txt
@@ -14,12 +14,19 @@ pico_add_subdirectory(pico_stdlib)
 pico_add_doxygen(${CMAKE_CURRENT_LIST_DIR})
 
 macro(pico_set_float_implementation TARGET IMPL)
+    # ignore
 endmacro()
 
 macro(pico_set_double_implementation TARGET IMPL)
+    # ignore
+endmacro()
+
+macro(pico_set_binary_type TARGET IMPL)
+    # ignore
 endmacro()
 
 macro(pico_set_boot_stage2 TARGET IMPL)
+    # ignore
 endmacro()
 
 set(PICO_HOST_DIR "${CMAKE_CURRENT_LIST_DIR}" CACHE INTERNAL "")

--- a/tools/elf2uf2/main.cpp
+++ b/tools/elf2uf2/main.cpp
@@ -59,16 +59,19 @@ struct address_range {
 
 typedef std::vector<address_range> address_ranges;
 
-#define MAIN_RAM_START 0x20000000u
-#define MAIN_RAM_END   0x20042000u
-#define FLASH_START    0x10000000u
-#define FLASH_END      0x15000000u
-#define XIP_SRAM_START 0x15000000u
-#define XIP_SRAM_END   0x15004000u
+#define MAIN_RAM_START        0x20000000u
+#define MAIN_RAM_END          0x20042000u
+#define FLASH_START           0x10000000u
+#define FLASH_END             0x15000000u
+#define XIP_SRAM_START        0x15000000u
+#define XIP_SRAM_END          0x15004000u
+#define MAIN_RAM_BANKED_START 0x21000000u
+#define MAIN_RAM_BANKED_END   0x21040000u
 
 const address_ranges rp2040_address_ranges_flash {
     address_range(FLASH_START, FLASH_END, address_range::type::CONTENTS),
-    address_range(MAIN_RAM_START, MAIN_RAM_END, address_range::type::NO_CONTENTS)
+    address_range(MAIN_RAM_START, MAIN_RAM_END, address_range::type::NO_CONTENTS),
+    address_range(MAIN_RAM_BANKED_START, MAIN_RAM_BANKED_END, address_range::type::NO_CONTENTS)
 };
 
 const address_ranges rp2040_address_ranges_ram {

